ims/einstein/operations.py

```diff
@@ -681,3 +682,26 @@ def run_script(self, img, script):
                                    constants.DEFAULT_SNAPSHOT_NAME)
             self.fs.remove_snapshot(ceph_img_name,
                                     constants.DEFAULT_SNAPSHOT_NAME)
+
+    # TODO: Write tests for this method before merging to main branch
+    @log
+    def get_iscsi_target(self, node_name):
+        try:
+            if not self.is_admin:
+                raise AuthorizationFailedException()
+            provisioned = False
+            provisioned_nodes = self.list_provisioned_nodes()['retval']
+            for node in provisioned_nodes:
+                if node_name == node[0]:
+                    provisioned = True
+            if not provisioned:
+                raise iscsi_exceptions.TargetDoesntExistException()
+            file_path = "/var/lib/tftpboot/" + node_name + ".ipxe"
+            with open(file_path, "r") as ipxe_file:
+                for line in ipxe_file:
+                    if "iscsi" in line:
+                        target = line.split("--keep ")[1][:-1]
+                        return self.__return_success({'iscsi_target': target})
```

Comment:
I guess you are just returning the string after --keep here ? Why not parse it and return a dict ? like 

```
{
    'ip': <>,
    'port': <>,
    'lun_id': <>,
    'target_name': <>
}
```